src/main/scala/za/co/absa/enceladus/kafkaplugin/KafkaPlugin.scala

```diff
@@ -17,55 +17,163 @@
+      metadata.put("sourceType", "MAINFRAME")
+      metadata.put("version", 1)
+      metadata.put("informationDate", "2020-02-21")
+      metadata.put("additionalInfo",arrayAddInfo)
+
+      genericRecord.put("datasetName", run.metadata.sourceApplication)
+      genericRecord.put("datasetVersion", run.metadata.version)
+      genericRecord.put("reportDate", params("reportDate"))
+      genericRecord.put("reportVersion", params("reportVersion").toInt)
+      genericRecord.put("runStatus", "Completed")
+      genericRecord.put("metadata",metadata)
+      genericRecord.put("checkpoints",arrayCheckpoint)
       genericRecord
+
     } catch {
       case NonFatal(ex) => throw new IllegalArgumentException("Unable to serialize a run object.", ex)
     }
 
-     */
   }
 
 }
 
 object KafkaPlugin extends ControlMetricsPluginFactory {
   private val logger = LoggerFactory.getLogger(this.getClass)
 
-  override def apply(conf: Config): ControlMetricsPlugin = {
+  def validateConfig(conf:Config,key:String): String ={
+    val status = conf.hasPath(key)
+    if (status== false) {
+      logger.info("value is empty")
+      System.exit(1)
+    }
+    conf.getString(key)
+  }
+
+  override def apply(conf: Config,schema: Schema,controlMeasure: ControlMeasure,valuegenericRecord:GenericRecord): Unit = {
 
     // Here an instance of Kafka Producer an be built based on the configuration passed as `conf`
 
     // Configuration can be something like this. Probably a validation is needed to ensure all required parameters have been passed.
 
-    /*
-    enceladus.kafka.schema.registry.url=http://zapalnrapp1079.corp.dsarena.com:8081
-    enceladus.kafka.bootstrap.servers=zapalnrapp1078.corp.dsarena.com:9092
-    enceladus.kafka.client.id=testInfo1
-    enceladus.kafka.topic.name=global.info
-      ...
-     */
+    val BOOTSTRAP_SERVERS_CONFIG = validateConfig(conf,"bootstrap.servers")
+    val SCHEMA_REGISTRY_URL_CONFIG= validateConfig(conf,"schema.registry.url")
+    val CLIENT_ID_CONFIG= validateConfig(conf,"client.id")
+    val KEY_SERIALIZER_CLASS_CONFIG= validateConfig(conf,"key.serializer")
+    val VALUE_SERIALIZER_CLASS_CONFIG= validateConfig(conf,"value.serializer")
+
+    val props = new Properties()
+    props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, BOOTSTRAP_SERVERS_CONFIG)
+    props.put(AbstractKafkaAvroSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG, SCHEMA_REGISTRY_URL_CONFIG)
+    props.put(CommonClientConfigs.CLIENT_ID_CONFIG, CLIENT_ID_CONFIG)
+    props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, KEY_SERIALIZER_CLASS_CONFIG)
+    props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, VALUE_SERIALIZER_CLASS_CONFIG)
 
     logger.info("Building Kafka plugin")
-    new KafkaPlugin()
+    //val genericRecord = new KafkaPlugin()
+
+    val producer = new KafkaProducer[GenericRecord,GenericRecord](props)
+
+    try {
+
+      val keySchemaString = "{\"type\": \"record\",\"name\": \"infoKey\",\"fields\":[{\"type\": \"string\",\"name\": \"key\"}]}}"
```

Comment:
```suggestion
      val keySchemaString = """{"type": "record", "name": "infoKey", "fields": [{"type": "string", "name": "key"}]}}"""
```

Might help with readability